### ISSUE TITLE
DefaultBackpropStrategy seeds loss.Backward with the loss VALUE instead of 1.0 (gradients scaled by loss magnitude)

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#872]
## Summary

`training.computeGradientsTensorCommon` (in `training/strategy_common.go`) seeds the loss node's backward pass with the **loss tensor value** rather than a ones tensor:

```go
// training/strategy_common.go
lossTensor, err := loss.Forward(ctx, output, batch.Targets)
...
lossGrads, err := loss.Backward(ctx, mode, lossTensor, output, batch.Targets)  // dOut = loss VALUE
```

`CrossEntropyLoss.Backward` (and `CrossEntropyLossOneHot.Backward`) multiply the computed `dL/dlogits` by this `dOut`:

```go
finalGradPredictions, err := m.engine.Mul(ctx, gradPredictions, dOut, nil)
```

The loss node's own doc comments say *"dOut is the gradient from the subsequent layer (usually 1.0 for loss)"*. For a scalar objective `L`, the correct seed is `dL/dL = 1`. Seeding with `L` scales **every** parameter gradient by the loss magnitude, i.e. the optimizer sees `L * ∇L` instead of `∇L`. This is an effective per-step learning-rate that rises and falls with the loss value.

## Reproduction (CPU, float32)

Two-row logits, well-conditioned:

```
logits = [[0.1, 0.2, -0.3], [0.5, -0.1, 0.0]], targets = [0, 2]
loss value = 1.1443887

CrossEntropyLoss.Backward(dOut = loss=1.1443887) -> dl row0 = [-0.36603466, 0.22784169, 0.13819297]
CrossEntropyLoss.Backward(dOut = 1.0)            -> dl row0 = [-0.31985170, 0.19909467, 0.12075702]
```

The `dOut=loss` result is exactly the correct `(softmax-onehot)/positions` times the loss value (1.1443887). Only the `dOut=1.0` row is the mathematically correct gradient.

## Impact

- All `DefaultBackpropStrategy`-driven training scales gradients by the per-step loss. Models still converge (it is a positive rescaling, effectively an adaptive LR), but the optimizer's nominal LR / clip thresholds do not mean what they appear to.
- It breaks **gradient equivalence across batching schemes**: a per-sample loop (one loss per sample) weights each sample's gradient by that sample's loss; a single batched backward (one mean loss for the batch) weights every sample by the batch-mean loss. The two are NOT equal when sample losses differ, even though the underlying math should be identical. This blocks batched-vs-per-sample gradient-parity validation in downstream batch-vectorization work (Wolf E13).

## Suggested fix

Seed the loss backward with a ones tensor matching the loss output shape (e.g. `[1]`), not the loss value:

```go
ones, err := tensor.New[T](lossTensor.Shape(), <ones>)
...
lossGrads, err := loss.Backward(ctx, mode, ones, output, batch.Targets)
```

This is a behavior change for existing training runs (gradient magnitudes drop by the loss factor), so it should land with a note for downstreams to re-tune LR/clip, and ideally a regression test asserting `loss.Backward` is seeded with 1.0.

## Environment

- zerfoo v1.50.0, ztensor v1.11.1
- Go 1.x, darwin/arm64 (also affects linux/arm64 GB10)
- Surfaced by Wolf CrossAsset batch-vectorization (E13 Stage 4) gradient-parity test.
